### PR TITLE
Feature/inventory system

### PR DESCRIPTION
**Pull request by AMastryukov**
- Add primary, secondary, melee, equipment and grenade slots
- Change relevant areas to use SlotType instead of HoldType
- Replace hardcoded HoldType with variable values
- Add inventory slot capacity logic
- Add random SMG/Shotgun on spawn (either one 50-50 chance)
- Remove redundant code from weapon implementations
- Adjust reload times to match animations better (when magazine is inserted)
- Update shop UI to reflect new slot behavior
- Update weapon selection UI for unlimited ammo weapons

### CODE REVIEW

File: code/player/Inventory.cs [hunk 1]
```diff
@@ -10,8 +10,8 @@ namespace TTTReborn.Player
     public partial class Inventory : BaseInventory
     {
         public readonly PerksInventory Perks;
-
         public readonly AmmoInventory Ammo;
+        public int[] SlotCapacity => new int[] { 1, 1, 1, 3, 3 };
```
Comment (Alf21) [inline code/player/Inventory.cs:14]:
```suggestion
        public readonly int[] SlotCapacity => new int[] { 1, 1, 1, 3, 3 };
```

File: code/player/Inventory.cs [hunk 2]
```diff
@@ -42,13 +42,13 @@ public override bool Add(Entity entity, bool makeActive = false)
         {
             TTTPlayer player = Owner as TTTPlayer;
 
-            if (IsCarryingType(entity.GetType()))
-            {
-                return false;
-            }
```
Comment (Alf21) [inline code/player/Inventory.cs:48]: Shouldn't this be kept if you trying to pickup something you already have that is not derived from `ICarriableItem`?

File: code/ui/alivehud/inventoryselection/InventorySelection.cs [hunk 3]
```diff
@@ -186,6 +186,11 @@ private int GetKeyboardNumberPressed(InputBuilder input)
 
         private static string FormatAmmo(TTTWeapon weapon, Inventory inventory)
         {
+            if (weapon.UnlimitedAmmo)
+            {
+                return $"{weapon.AmmoClip}";
+            }
```
Comment (Alf21) [inline code/ui/alivehud/inventoryselection/InventorySelection.cs:192]: Good catch!

File: code/items/IBuyableItem.cs [hunk 4]
```diff
@@ -9,21 +11,20 @@ public struct ShopItemData
         public string Name;
         public string Description;
         public int Price;
+        public SlotType SlotType;
```
Comment (Alf21) [inline code/items/IBuyableItem.cs:14]: That's the only change I don't like in this PR, as Perks are `IBuyableItems` as well, but never have a `SlotType`

**Comment by mzegar** [comment]
Don't have anything to add, looks fine to me, check out Alf comments :D

File: code/player/Inventory.cs (hunk 2 above)
Comment (AMastryukov) [reply code/player/Inventory.cs:48]: Damn, good catch. I will revert.

File: code/items/IBuyableItem.cs (hunk 4 above)
Comment (AMastryukov) [reply code/items/IBuyableItem.cs:14]: I agree, I don't like this change either.

We should discuss what alternatives exist to this approach of checking if players are able to purchase a weapon/equipment when there is no slot available.

File: code/player/Inventory.cs (hunk 2 above)
Comment (AMastryukov) [reply code/player/Inventory.cs:48]: Reverted

File: code/items/IBuyableItem.cs (hunk 4 above)
Comment (Alf21) [reply code/items/IBuyableItem.cs:14]: `SlotType` still could be a part of `ICarriableItem`, then you just need to check for it explicitly.

**Review by Alf21 — APPROVED** [review]
Nice improvements :)

**Pull request merged**